rust/derive/src/stringenum.rs

```diff
@@ -0,0 +1,127 @@
+    let input = parse_macro_input!(input as DeriveInput);
+    let name = input.ident;
+    let mut values = Vec::new();
+    let mut names = Vec::new();
+    let mut fields = Vec::new();
+
+    if let syn::Data::Enum(ref data) = input.data {
+        let mut default_seen = false;
+        for (_, v) in (&data.variants).into_iter().enumerate() {
+            if let Some((_, val)) = &v.discriminant {
+                let fname = transform_name(&v.ident.to_string());
+                names.push(fname);
+                fields.push(v.ident.clone());
+                if let syn::Expr::Lit(l) = val {
+                    if let syn::Lit::Int(li) = &l.lit {
+                        if let Ok(value) = li.base10_parse::<T>() {
+                            values.push(value);
+                        } else {
+                            panic!("EnumString requires explicit {}", ustr);
+                        }
+                    } else {
+                        panic!("EnumString requires explicit literal integer");
+                    }
+                } else {
+                    panic!("EnumString requires explicit literal");
+                }
+            } else {
+                if let syn::Fields::Unnamed(f) = &v.fields {
+                    if default_seen || f.unnamed.len() != 1 {
+                        panic!("EnumString requires explicit values or one Unknown({})", ustr);
+                    }
+                    if v.ident.to_string() != "Unknown" {
+                        panic!("EnumString default case must be Unknown({})", ustr);
+                    }
+                    if let syn::Type::Path(p) = &f.unnamed[0].ty {
+                        if p.path.segments.len() != 1 {
+                            panic!("EnumString requires explicit values or one Unknown({})", ustr);
+                        }
+                        if p.path.segments[0].ident.to_string() != ustr {
+                            panic!("EnumString default case must be Unknown({})", ustr);
+                        }
+                    } else {
+                        panic!("EnumString requires explicit values or one Unknown({})", ustr);
+                    }
+                    default_seen = true;
+                } else {
+                    panic!("EnumString requires explicit values or one Unknown({})", ustr);
+                }
+            }
+        }
+    } else {
+        panic!("EnumString can only be derived for enums");
+    }
+
+    let utype_str = syn::Ident::new(&ustr, proc_macro2::Span::call_site());
+
+    let expanded = quote! {
+        impl #name {
+            pub(crate) fn from_u(v: #utype_str) -> Self {
+                match v {
+                    #( #values => #name::#fields ,)*
+                    _ => #name::Unknown(v),
+                }
+            }
+            pub(crate) fn into_u(&self) -> #utype_str {
```

Comment:
Why not just `from` and `into`?

Reply:
Because they are standard keywords ?

Reply:
Maybe into_u8 and from_u8 to be a little more verbose given the name of the proc macro.

Also, any other enums we can implement this on to show its broader use case?

Reply:
>  any other enums we can implement this on to show its broader use case?

For example http2.errorcode, http2.frametype (checking all http2-specific keywords)